banzai_nres/traces.py

```diff
@@ -9,43 +9,42 @@
 import logging
 from scipy import ndimage
 import numpy as np
-from banzai_nres.fitting import fit_smooth_spline
+from numpy.polynomial.polynomial import Polynomial, polyfit
 
 logger = logging.getLogger('banzai')
 
 # Minimum separation between peaks in the image that could be separate traces
 MIN_TRACE_SEPARATION = 10
 
 # Cutoff in signal-to-noise to stop following a trace
-SIGNAL_TO_NOISE_TRACING_CUTOFF = 50
+SIGNAL_TO_NOISE_TRACING_CUTOFF = 10
 
-# Minimum half width of a feature to be considered a trace
-MIN_TRACE_HALF_WIDTH = 50
+# Maximum degree of polynomial which describes the y position of the trace as a function of pixel
+POLY_DEGREE = 5
 
 # The final trace will be +- this from the center in the y-direction
 TRACE_HALF_HEIGHT = 6
 
 
 def find_y_center(y, indices, weights=None):
     if weights is None:
-        weights = np.ones(y.shape, dtype=y.dtype)
+        weights = 1
     centers = (y * indices * weights).sum(axis=0) / (indices * weights).sum(axis=0)
-    # Errors are sqrt sum of the weights
-    errors = np.sqrt((indices * weights).sum(axis=0))
+    # Errors are sqrt sum of the weights squared
+    errors = np.sqrt((indices * weights ** 2).sum(axis=0))
     return centers, errors
 
```

Comment:
If f = w1*x1 + w2*x2, then var(f) = w1**2 var(x1) + w2**2 var(x2) 
etc...

Reply:
So this was based on the B-spline weights. We were using linear weights, see smoothing factor here: https://docs.scipy.org/doc/scipy/reference/generated/scipy.interpolate.SmoothBivariateSpline.html

For a polynomial we should use the normal inverse variance weighting. 

Though we may want to use iterative reweighting to mitigate outliers. https://en.wikipedia.org/wiki/Iteratively_reweighted_least_squares

Reply:
Ok